### PR TITLE
test: phoneInput when only one country is passed

### PR DESCRIPTION
**Pull request by gtallesb**
fix: #1701 

## Changes proposed in this PR:
-add if block to the useCountries hook
-add test to check when an empty array is passed in countries prop
-add test to check when an array containing empty string is passed
-add test to check if an array contains an invalid country code

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/PhoneInput/hooks/useCountries.js [hunk 1]
```diff
@@ -3,9 +3,13 @@ import allCountries from '../countries';
 
 export default function useCountries(countries) {
     return useMemo(() => {
-        if (countries.length === 0) {
+        if ((countries.length === 0)) {
```
Comment (yvmunayev) [inline src/components/PhoneInput/hooks/useCountries.js:6]: lint error

```
/home/circleci/repo/src/components/PhoneInput/hooks/useCountries.js
   6:13  error    Replace `(countries.length·===·0)` with `countries.length·===·0`                                                                                              prettier/prettier
```

File: src/components/PhoneInput/__tests__/phoneinput.spec.js [hunk 2]
```diff
@@ -32,4 +32,16 @@ describe('<PhoneInput />', () => {
...
     });
+
+    it('should render the dropdown option when an array containing empty string is passed', () => {
+        const countries = [''];
+        const wrapper = mount(<PhoneInput countries={countries} />);
+        expect(wrapper.find(StyledTrigger).prop('onClick')).toBeDefined;
+    });
+
+    it('should render the dropdown option when an array containing invalid country code is passed', () => {
+        const countries = [!(/^w{2}$/)];
```
Comment (yvmunayev) [inline src/components/PhoneInput/__tests__/phoneinput.spec.js:43]: lint error

  ```
39:9   error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  43:29  error  Replace `(/^w{2}$/)` with `/^w{2}$/`                                   prettier/prettier
  45:9   error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

File: src/components/PhoneInput/hooks/useCountries.js [hunk 3]
```diff
@@ -3,9 +3,13 @@ import allCountries from '../countries';
...
             return allCountries;
         }
+        if ((allCountries.filter(country => countries.includes(country.isoCode))).length === 0) {
```
Comment (yvmunayev) [inline src/components/PhoneInput/hooks/useCountries.js:9]: error lint

`
   9:12  error    Replace `((allCountries.filter(country·=>·countries.includes(country.isoCode)))` with `(allCountries.filter(country·=>·countries.includes(country.isoCode))`  prettier/prettier`

File: src/components/PhoneInput/hooks/useCountries.js [hunk 4]
```diff
@@ -3,9 +3,13 @@ import allCountries from '../countries';
...
+            console.warn('You have entered an invalid country code or empty input');
```
Comment (yvmunayev) [inline src/components/PhoneInput/hooks/useCountries.js:10]: no console log

File: src/components/PhoneInput/hooks/useCountries.js [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
 export default function useCountries(countries) {
     return useMemo(() => {
         if (countries.length === 0) {
             return allCountries;
         }
+        if (allCountries.filter(country => countries.includes(country.isoCode)).length === 0) {
+            customWarning();
+            return allCountries;
+        }
         return allCountries.filter(country => countries.includes(country.isoCode));
```
Comment (LeandroTorresSicilia) [inline src/components/PhoneInput/hooks/useCountries.js:14]: here the same iteration occurs twice, instead you can:
```
const filteredConuntries = allCountries.filter(country => countries.includes(country.isoCode));
if (filteredConuntries.length === 0) {
    customWarning();
    return allCountries;
}
return filteredConuntries;
```

**Review by LeandroTorresSicilia — APPROVED** [review]
